### PR TITLE
Custom image upload for page and board icon

### PR DESCRIPTION
**Pull request by Devorein**
### <samp>🤖 Generated by Copilot at 3ce86bd</samp>

This pull request adds the feature of custom image icons for pages and boards, and refactors some components to support this feature. It introduces a new `PageHeaderIcon` component that allows the user to choose between an emoji or an image as the icon, and uses it in the `PageHeader` and `blockIconSelector` components. It also extracts the image upload logic to a separate `ImageUploadButton` component, and updates the `ImageSelector`, `PageIcon`, and `PageTreeItem` components to handle image urls as icons. It applies some custom styles to the `EmojiPicker` component and the `img` element inside the `EmojiIcon` component to improve the appearance of the icons.

### WHY
[Card](https://app.charmverse.io/charmverse/page-059699641486480326)

### CODE REVIEW

File: components/common/BoardEditor/focalboard/src/widgets/emojiPicker.scss [hunk 1]
```diff
@@ -18,7 +18,21 @@
   }
 }
 
+div[role="tabpanel"] .EmojiPicker .emoji-mart {
```
Comment (Devorein) [inline components/common/BoardEditor/focalboard/src/widgets/emojiPicker.scss:21]: This is to remove background inconsistencies between the tab panel and emoji-mart

File: components/common/ImageSelector/ImageUploadButton.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        disabled={isUploading}
+        component='label'
+        variant={variant}
+        {...props}
+      >
+        Choose an image
+        <input
+          disabled={isUploading}
+          type='file'
+          style={{
```
Comment (Devorein) [inline components/common/ImageSelector/ImageUploadButton.tsx:37]: Couldn't get it working with <input hidden /> ... thats why the complex `style`, when it is inside the bangle-dev Menu component

File: components/common/PageLayout/components/PageIcon.tsx [hunk 3]
```diff
@@ -91,25 +92,34 @@ export function PageIcon({ icon, isEditorEmpty, isLinkedPage = false, pageType,
...
+      iconComponent = <MessageOutlinedIcon />;
     } else if (isEditorEmpty) {
-      icon = <EmptyPageIcon />;
+      iconComponent = <EmptyPageIcon />;
     } else {
-      icon = <FilledPageIcon />;
+      iconComponent = <FilledPageIcon />;
     }
+  } else if (typeof icon === 'string' && icon.startsWith('http')) {
+    iconComponent = (
```
Comment (Devorein) [inline components/common/PageLayout/components/PageIcon.tsx:110]: Making sure mentions show the correct icon

**Review by mattcasey — APPROVED** [review]

File: components/common/ImageSelector/ImageUploadButton.tsx [hunk 4]
```diff
@@ -12,14 +13,18 @@ export function ImageUploadButton({
...
+  fileSizeLimit,
   ...props
 }: {
   isUploading: boolean;
   setIsUploading: (isUploading: boolean) => void;
   setImage: (image: string) => void;
   uploadDisclaimer?: string;
   variant?: ButtonProps['variant'];
+  // file size limit in megabytes
+  fileSizeLimit?: number;
```
Comment (msadura) [inline components/common/ImageSelector/ImageUploadButton.tsx:25]: name it `fileSizeLimitMB` and you will not need that comment ;)

**Review by msadura — CHANGES_REQUESTED** [review]

File: components/common/ImageSelector/ImageUploadButton.tsx (hunk 4 above)
Comment (Devorein) [reply components/common/ImageSelector/ImageUploadButton.tsx:25]: Good point :D

File: components/common/ImageSelector/ImageUploadButton.tsx [hunk 5]
```diff
@@ -51,6 +56,11 @@ export function ImageUploadButton({
             setIsUploading(true);
             const firstFile = e.target.files?.[0];
             if (firstFile) {
+              if (firstFile.size > (fileSizeLimit ?? 10) * 1024 * 1024) {
```
[the remark this answers is not in the document]
Comment (Devorein) [reply components/common/ImageSelector/ImageUploadButton.tsx:59]: Ow crap, I didn't even realize that hook existed 🤦🏼‍♂️ Thank you for sharing this. I've refactored the code and removed the unnecessary styling on the input element as well.

**Comment by msadura** [comment]
nice work with wrapping emoji picker lib with MUI tabs. One detail - can you make emojis use our branding color instead of purple? (I believe even hardcoded css color will be ok)

Screenshot 2023-07-26 at 12 36 02

**Pull request merged**